systemtest/src/main/java/io/strimzi/systemtest/performance/utils/TopicOperatorPerformanceUtils.java

```diff
@@ -0,0 +1,272 @@
+                } catch (Exception e) {
+                    LOGGER.error("Error while checking for topic deletion: ", e);
+                    return e.getMessage().contains("Not Found") || e.getMessage().contains("the server doesn't have a resource type");
+                }
+            });
+    }
+
+    /**
+     * Manages the full lifecycle of Kafka topics concurrently using a fixed thread pool.
+     * This method processes creation, modification, and deletion for each topic in separate threads,
+     * ensuring all topics are handled simultaneously across available CPU resources.
+     *
+     * +-----------------------------------------------------------------------------+
+     * |                            processAllTopicsConcurrently                     |
+     * +-----------------------------------------------------------------------------+
+     * | +-------------+       +-------------+             +-------------+           |
+     * | | Topic 1     |       | Topic 2     |     ...     | Topic N     |           |
+     * | | +---------+ |       | +---------+ |             | +---------+ |           |
+     * | | | Thread  | |       | | Thread  | |             | | Thread  | |           |
+     * | | |         | |       | |         | |             | |         | |           |
+     * | | | Creation| |       | | Creation| |             | | Creation| |           |
+     * | | | Update  | |       | | Update  | |             | | Update  | |           |
+     * | | | Deletion| |       | | Deletion| |             | | Deletion| |           |
+     * | | +---------+ |       | +---------+ |             | +---------+ |           |
+     * | +-------------+       +-------------+             +-------------+           |
+     * +-----------------------------------------------------------------------------+
+     *
+     * @param testStorage       An instance of TestStorage containing configuration and state needed for topic operations.
+     * @param numberOfTopics    The number of Kafka topics to be processed.
+     */
+    public static void processAllTopicsConcurrently(TestStorage testStorage, int numberOfTopics) {
+        final int availableCPUs = Math.max(1, Runtime.getRuntime().availableProcessors());
+        final ExecutorService executor = Executors.newFixedThreadPool(availableCPUs);
+
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+
+        for (int topicIndex = 0; topicIndex < numberOfTopics; topicIndex++) {
+            final int finalTopicIndex = topicIndex;
+            CompletableFuture<Void> future = CompletableFuture.runAsync(() -> performFullLifecycle(finalTopicIndex, testStorage), executor);
+            futures.add(future);
+        }
+
+        // Wait for all topics to complete their lifecycle
+        CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).join();
+        LOGGER.info("All topic lifecycles completed.");
+
+        executor.shutdown();
```

Comment:
This may not be enough in some situations. Something like this is better:

```sh
        try {
            executor.shutdown();
            executor.awaitTermination(timeoutMs, TimeUnit.MILLISECONDS);
        } catch (InterruptedException e) {
            if (!executor.isTerminated()) {
                executor.shutdownNow();
            }
        }
```

Reply:
Sure :) I would add this thanks